service/history/replication/raw_task_converter.go

```diff
@@ -295,37 +292,25 @@ func convertSyncHSMReplicationTask(
 
 func convertSyncVersionedTransitionTask(
 	ctx context.Context,
-	shardContext shard.Context,
 	taskInfo *tasks.SyncVersionedTransitionTask,
-	shardID int32,
-	workflowCache wcache.Cache,
-	eventBlobCache persistence.XDCCache,
-	replicationCache ProgressCache,
 	targetClusterID int32,
-	executionManager persistence.ExecutionManager,
-	logger log.Logger,
+	converter *syncVersionedTransitionTaskConverter,
 ) (*replicationspb.ReplicationTask, error) {
+	converter.logger = log.With(
+		converter.logger,
+		tag.WorkflowNamespaceID(taskInfo.NamespaceID),
+		tag.WorkflowID(taskInfo.WorkflowID),
+		tag.WorkflowRunID(taskInfo.RunID),
+		tag.TaskKey(taskInfo.GetKey()),
+		tag.Task(taskInfo),
+	)
```

Comment:
Will this keep adding tags to `converter.logger` as it processes tasks?

Reply:
Good point. Removed this part and add tag in each indivitual log call.